underfs/swift/src/main/java/tachyon/underfs/swift/direct/SwiftDirectUnderFileSystem.java

```diff
@@ -0,0 +1,400 @@
+    } else {
+      config.setAuthenticationMethod(AuthenticationMethod.TEMPAUTH);
+    }
+
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.configure(SerializationConfig.Feature.WRAP_ROOT_VALUE, true);
+    mContainerName = containerName;
+    mAccount = new AccountFactory(config).createAccount();
+    mAccess = mAccount.authenticate();
+    LOG.debug("Check if container: {} exists ", containerName);
+    Container containerObj = mAccount.getContainer(containerName);
+    if (!containerObj.exists()) {
+      containerObj.create();
+    }
+    mContainerPrefix = Constants.HEADER_SWIFT + mContainerName
+        + PATH_SEPARATOR;
+  }
+
+  @Override
+  public void close() throws IOException {
+    LOG.debug("close");
+  }
+
+  @Override
+  public void connectFromMaster(TachyonConf conf, String hostname) {
+    LOG.debug("connect from master");
+  }
+
+  @Override
+  public void connectFromWorker(TachyonConf conf, String hostname) {
+    LOG.debug("connect from worker");
+  }
+
+  @Override
+  public OutputStream create(String path) throws IOException {
+    LOG.debug("Create method: " + path);
+    String newPath = path.substring(Constants.HEADER_SWIFT.length());
+    SwiftDirectOutputStream out = SwiftDirectClient.PUT(mAccess, newPath);
+    return out;
+  }
+
+  @Override
+  public OutputStream create(String path,
+      int blockSizeByte) throws IOException {
+    LOG.warn("Create with block size is not supported"
+        + "with SwiftDirectUnderFileSystem. Block size will be " + "ignored.");
+    return create(path);
+  }
+
+  @Override
+  public OutputStream create(String path, short replication, int blockSizeByte) throws IOException {
+    LOG.warn("Create with block size and replication is not"
+        + "supported with SwiftDirectUnderFileSystem."
+        + " Block size and replication will be ignored.");
+    return create(path);
+  }
+
+  /* (non-Javadoc)
+   * @see tachyon.underfs.UnderFileSystem#delete(java.lang.String, boolean)
+   *
+   * recursive parameter is not being used for delete.
+   * The reason is that each path is treated as a single object, which
+   * was created with 'create' method.
+   *
+   * Method return 'true' always.
+   */
+  @Override
+  public boolean delete(String path, boolean recursive) throws IOException {
+    LOG.debug("Delete method: " + path + ",recursive: " + recursive);
+    String strippedPath = stripPrefixIfPresent(path);
+    LOG.debug("Going to retrieve container: " + mContainerName);
+    Container c = mAccount.getContainer(mContainerName);
+    LOG.debug("Going to retrieve an object: " + strippedPath);
+    StoredObject so = c.getObject(strippedPath);
+    LOG.debug("Checking if object exists in Swift. If so, delete it");
+    if (so.exists()) {
+      LOG.debug("Object exists: " + so.getBareName());
+      so.delete();
+    }
+    return true;
+  }
+
+  @Override
+  public boolean exists(String path) throws IOException {
+    LOG.debug("Exists: " + path);
```

Comment:
again, I think we should either clean this up, or make it
`LOG.debug("Exists: {}", path);` so when debug is off, it won't cause string calculation which is fairly expensive in Java land.